src/main.rs

```diff
@@ -844,6 +927,7 @@ type LavaMap = Arc<Mutex<BTreeMap<(String, String), Arc<ThrottledLava>>>>;
 lazy_static! {
     static ref LAVA_MAP: LavaMap = Arc::new(Mutex::new(BTreeMap::new()));
     static ref MAX_CONCURRENT_REQUESTS: Arc<Mutex<usize>> = Arc::new(Mutex::new(20));
+    static ref UPLOAD_SERVER: Arc<Mutex<UploadServer>> = Default::default();
```

Comment:
rather then having a global static variable why  pass a reference to the Run struct ; that avoids the global variable  and potentlaly a lot of the lock().unwrap() dances (though that could be sorted out by internal mutability as wel..

Tbh it would probably be nicer if  the Runs each simply had a copy or a registration object from the factory